### PR TITLE
Add custom importers for importing types/values

### PR DESCRIPTION
**Pull request by phated**
**Alright, I think this is ready to get some serious feedback. Promoting from draft!**

Building on the excellent work by @devongovett and @motiz88, I've been trying to implement the requested changes in #352 (specifically https://github.com/reactjs/react-docgen/pull/352#issuecomment-499248734)

What I've done
* Made the "importer" configurable. It is passed as an argument to the `parse` method, but if using the `defaultParse` from `main.js`, it is an additional option that can be specified. **This is to make everything backwards compatible.**
* Made the default importer a no-op, which might result in some extra function calls during resolution, but keeps all behavior completely the same unless someone explicitly overrides it. **This is to make everything backwards compatible.**
* Threaded the importer throughout the codebase.
* Added tests for a "mockImporter" throughout the entire test suite.
* Renamed the importer added in #352 to `fsImporter` and wrapped it in a factory function so the module lookup and cache could be configured. This also allows caching to be done for entire run of react-docgen (if you have a way to reuse your cache, it can even be done between runs).

Stuff I still need to do:
- [ ] Add tests for `makeFsImporter`
- [ ] Add docs for importers

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -43,11 +43,12 @@
   "dependencies": {
     "@babel/core": "^7.7.5",
     "@babel/runtime": "^7.7.6",
-    "ast-types": "^0.13.2",
+    "@motiz88/ast-types": "^0.13.3",
```
Comment (phated) [inline package.json:46]: Due to an upstream patch that hasn't landed yet. I pinged the thread last week.

**Comment by phated** [comment]
I might have run into a problem when testing this against some local code today. Need to dig in further.

File: package.json (hunk 1 above)
Comment (phated) [reply package.json]: From what I can tell, ast-types 0.14 actually fixes the flow stuff.

**Comment by phated** [comment]
Alright, I think this is ready for an initial review while I continue the follow up work of adding test for `makeFsImporter` and document importers.

What do you think @danez?

**Comment by shilman** [comment]
Heroic work @phated 🎖️ 🏅 🥇

**Comment by phated** [comment]
It's probably worth mentioning; I bootstrapped this into a large project that I'm working on and it works splendidly!

**Comment by kivervinicius** [comment]
@phated wow!

**Comment by kivervinicius** [comment]
@phated can i use this with you repository? In npm

**Comment by danez** [comment]
I will merge now and release an alpha version with it. Feel free to create follow up PRs for the tests, bugfixes etc.

**Comment by stu039** [comment]
@danez - Any documentation on usage of this please? Using imported proptypes has been an issue for us for a while. If there is no documentation showing how to enable this in the alpha version, we may have to move away from docgen.

**Comment by danez** [comment]
@stu039 see https://github.com/reactjs/react-docgen/discussions/507 the cli needs to be adapted for this to work. There might be also still bugs.

**Comment by dozortsev** [comment]
@danez @phated I guess note that imported types aren't supported is redundant, right?

https://github.com/reactjs/react-docgen/blob/master/README.md#flow-and-typescript-support

**Pull request merged**